ops/internal/manage/fetch_onchain.go

```diff
@@ -0,0 +1,128 @@
+package manage
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/ethereum-optimism/optimism/op-fetcher/pkg/fetcher/fetch"
+	"github.com/ethereum-optimism/optimism/op-fetcher/pkg/fetcher/fetch/script"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/config"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/paths"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/sync/errgroup"
+)
+
+// FetchChains fetches chain configurations for specified chain IDs or all chains if none specified
+func FetchChains(lgr log.Logger, wd string, l1RpcUrls []string, chainIdStrs []string) (map[uint64]script.ChainConfig, error) {
```

Comment:
Why are the chain IDs passed in as strings? They should be a `uint64` (or, for a future refactor, `eth.ChainID`) in public methods. The CLI should be the only thing that deals with string chain IDs.